apm-agent-plugins/apm-redis-plugin/apm-lettuce-plugin/src/main/java/co/elastic/apm/agent/redis/lettuce/Lettuce34Instrumentation.java

```diff
@@ -0,0 +1,53 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.redis.lettuce;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import io.lettuce.core.protocol.RedisCommand;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers.classLoaderCanLoadClass;
+
+public abstract class Lettuce34Instrumentation extends ElasticApmInstrumentation {
```

Comment:
If you want to make consistent with the Elasticsearch client instrumentation, I specified the version in the package name and not in the class name, eg: `co.elastic.apm.agent.redis.lettuce.v3_4.LettuceInstrumentation`

Reply:
Because Lettuce 3 and 4 use a different base package (`com.lambdaworks.redis` vs `io.lettuce`) I don't need another maven module which makes it easier to share code.

Reply:
Still, the version can be in the package. But not very important.